clients/node/client-rds-data-node/protocol/AwsRestJson1_1.ts

```diff
@@ -0,0 +1,214 @@
+import {
+  ColumnMetadata,
+  ExecuteStatementRequest,
+  ExecuteStatementResponse,
+  Field,
+  SqlParameter
+} from "../models/com/amazon/rdsdataservice";
+import { HttpRequest, HttpResponse } from "@aws-sdk/protocol-http";
+
+export function executeStatementAwsRestJson1_1Serialize(
+  input: ExecuteStatementRequest
+): HttpRequest {
+  let body: any = {};
+  if (input.resourceArn !== undefined) {
+    body.resourceArn = input.resourceArn;
+  }
+
+  if (input.secretArn !== undefined) {
+    body.secretArn = input.secretArn;
+  }
+
+  if (input.sql !== undefined) {
+    body.sql = input.sql;
+  }
+
+  if (input.database !== undefined) {
+    body.database = input.database;
+  }
+
+  if (input.schema !== undefined) {
+    body.schema = input.schema;
+  }
+
+  if (input.parameters !== undefined) {
+    body.parameters = sqlParameterListAwsRestJson1_1Serialize(input.parameters);
+  }
+
+  if (input.transactionId !== undefined) {
+    body.transactionId = input.transactionId;
+  }
+
+  if (input.includeResultMetadata !== undefined) {
+    body.includeResultMetadata = input.includeResultMetadata;
+  }
+
+  if (input.continueAfterTimeout !== undefined) {
+    body.continueAfterTimeout = input.continueAfterTimeout;
+  }
+
+  return new HttpRequest({
+    body: JSON.stringify(body),
+    path: "/execute",
+    method: "POST",
+    protocol: "https:",
+    headers: {
+      "Content-Type": "application/json"
+    }
+  });
+}
+
+export function executeStatementAwsRestJson1_1Deserialize(
```

Comment:
Deserializers should be async functions that return a promise according to [our definition](https://github.com/AllanFly120/aws-sdk-js-v3/blob/c60241b224457f94c33f6b3dc47ba5f9a0a175e6/packages/types/src/unmarshaller.ts#L26)